src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -1004,12 +1004,25 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
 
         case NI_Vector128_ConvertToInt32:
         case NI_Vector256_ConvertToInt32:
+        case NI_Vector512_ConvertToInt32:
         {
             assert(sig->numArgs == 1);
             assert(simdBaseType == TYP_FLOAT);
 
-            intrinsic = (simdSize == 32) ? NI_AVX_ConvertToVector256Int32WithTruncation
-                                         : NI_SSE2_ConvertToVector128Int32WithTruncation;
+            switch (simdSize)
+            {
+                case 16:
+                    intrinsic = NI_SSE2_ConvertToVector128Int32WithTruncation;
+                    break;
+                case 32:
+                    intrinsic = NI_AVX_ConvertToVector256Int32WithTruncation;
+                    break;
+                case 64:
+                    intrinsic = NI_AVX512F_ConvertToVector512Int32WithTruncation;
+                    break;
+                default:
+                    unreached();
+            }
```

Comment:
nit: We might not want to switch on the size because we sometimes see `simdSize == 8` and `simdSize == 12` for some nodes.

I don't think we currently have any for `ConvertTo`, but its probably better to be "safe" and do `if (simdSize == 64) { } else if (simdSize == 32) { } else { }` like the other paths are doing.

Reply:
@tannergooding - so for `simdSize == 12` and `simdSize == 8` we will end up with `NI_SSE2_ConvertToVector128Int32WithTruncation` which is ok?

@anthonycanino - can you address this?

Reply:
Right. We won't see it today so its not required to fix it "right now". But there are some optimizations we'd like to do in the future that would need it so we'd want to at least fix it at that time